internal/templates/containercapabilities/template.go

```diff
@@ -0,0 +1,218 @@
+package containercapabilities
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"golang.stackrox.io/kube-linter/internal/check"
+	"golang.stackrox.io/kube-linter/internal/diagnostic"
+	"golang.stackrox.io/kube-linter/internal/matcher"
+	"golang.stackrox.io/kube-linter/internal/objectkinds"
+	"golang.stackrox.io/kube-linter/internal/templates"
+	"golang.stackrox.io/kube-linter/internal/templates/containercapabilities/internal/params"
+	"golang.stackrox.io/kube-linter/internal/templates/util"
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	reservedCapabilitiesAll             = "all"
+	matchLiteralReservedCapabilitiesAll = "(?i)" + reservedCapabilitiesAll
+)
+
+var (
+	literalReservedCapabilitiesAllMatcher func(string) bool
+)
+
+func checkCapabilityDropList(
+	containerName string,
+	paramCapMatchers map[string]func(string) bool,
+	scCaps *v1.Capabilities,
+	forbidAll bool,
+) *diagnostic.Diagnostic {
+	if forbidAll {
+		// Specified to drop all. Instead of trying to find a match to this paramCap, we should
+		// make sure the special ReservedCapabilitiesAll is found in the DROP list as well
+		for _, scCap := range scCaps.Drop {
+			if literalReservedCapabilitiesAllMatcher(string(scCap)) {
+				// Found "all" in drop list as well
+				return nil
+			}
+		}
+		return &diagnostic.Diagnostic{
+			Message: fmt.Sprintf(
+				"container %q has DROP capabilities: %q, but in fact all capabilities are required to be dropeed",
+				containerName,
+				scCaps.Drop),
+		}
+	}
+
+	// Every forbidden capability specified by param should be found in the DROP list
+	for paramCap, paramCapMatcher := range paramCapMatchers {
+		var found bool
+		for _, scCap := range scCaps.Drop {
+			// User can specify to drop "all" under containers as well
+			if paramCapMatcher(string(scCap)) || literalReservedCapabilitiesAllMatcher(string(scCap)) {
+				// This forbidden cap exists in the DROP list. Check for the next forbidden cap
+				found = true
+				break
+			}
+		}
+		if !found {
+			return &diagnostic.Diagnostic{
+				Message: fmt.Sprintf("container %q has DROP capabilities: %q, but does not drop "+
+					"capability %q which is required",
+					containerName,
+					scCaps.Drop,
+					paramCap),
+			}
+		}
+	}
+	return nil
+}
+
+func checkCapabilityAddList(
```

Comment:
You can make this function take in a pointer to the output slice (ie, a `*[]diagnostic.Diagnostic`) and just append your findings to the slice.